plugins/urlscan/komand_urlscan/actions/search/action.py

```diff
@@ -37,27 +38,48 @@ def run(self, params={}):
                 )
             search_query = f'page.domain:"{query}"'
 
-        query_params = [
-            f"q={search_query}",
-            f"size={str(params.get(Input.SIZE, 100))}",
-            f"offset={str(params.get(Input.OFFSET, 0))}",
-            f'sort={params.get(Input.SORT, "_score")}',
-        ]
-
-        url = f'{self.connection.server}/search/?{"&".join(query_params)}'
-        self.logger.info(url)
-
-        try:
-            response = requests.get(url, headers=self.connection.headers)
-        except Exception as e:
-            raise PluginException(cause="Something went wrong during the request. ", assistance=e)
-
-        try:
-            output = response.json()
-        except json.decoder.JSONDecodeError:
-            raise PluginException(
-                cause="Received an unexpected response from the Urlscan API. ",
-                assistance=f"(non-JSON or no response was received). Response was: {response.text}",
-            )
-
-        return output
+        search_after = None
+        results = []
+        size = 10000
+        has_more = False
+        for i in range(1, 9999):
```

Comment:
Refactor this to be a while loop. Don't do for range if you don't know the end case for the range.

Reply:
@jmcadams-r7 We will change this :) We implement here `for` instead `while`, because in while if there will be a problem on a server sometimes we can get infinite loop. With `for` we know that it stops after 9999 pages. This value is big enough to return many objects.

Reply:
Ok, no need to change. That makes sense.